src/meta_memcache/base/base_cache_pool.py

```diff
@@ -182,13 +195,30 @@ def _conn_send_cmd(
     def _conn_recv_response(
         self,
         conn: MemcacheSocket,
+        command: MetaCommand,
         flags: Optional[Set[Flag]] = None,
     ) -> MemcacheResponse:
         """
         Read response on a connection
         """
+        # No-reply flag behavior is hard to implement, so this
+        # is implemented client side, recording how many responses
+        # to ignore in client-side. See:
+        # https://github.com/memcached/memcached/issues/868
         if flags and Flag.NOREPLY in flags:
+            if command != MetaCommand.META_GET:
+                self._no_reply_cmds += 1
             return Success(flags=set([Flag.NOREPLY]))
+        while self._no_reply_cmds > 0:
```

Comment:
Who is possible to have `self._no_reply_cmds` > 0 if the part where is increased, line 210, finishes with a return statement?

Reply:
The first command that doesn't use `NOREPLY` will read the results for commands sent with NOREPLY and drop those, so it can get the reply it wants.